k9mail-library/src/main/java/com/fsck/k9/mail/store/webdav/DataSet.java

```diff
@@ -52,10 +52,11 @@ public void finish() {
      */
     public Map<String, String> getSpecialFolderToUrl() {
         // We return the first (and only) map
-        for (Map<String, String> folderMap : mData.values()) {
-            return folderMap;
+        if (mData.size() == 1) {
```

Comment:
This changes the behavior if `mData` contains more than one element. It shouldn't according to the comment, but from glancing at the code it's not immediately obvious that comment is always true.

I'd change the code to the following:

```java
return mData.isEmpty() ? Collections.<String, String>emptyMap() : mData.values().iterator().next();
```